src/hsm.rs

```diff
@@ -69,11 +69,63 @@ pub enum HsmError {
     Version,
 }
 
+pub struct Alphabet {
+    chars: Vec<char>,
+}
+
+impl Default for Alphabet {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl Alphabet {
+    pub fn new() -> Self {
+        let mut chars: Vec<char> = Vec::new();
+
+        for i in b'a'..=b'z' {
+            chars.push(char::from(i));
+        }
+
+        for i in b'A'..=b'Z' {
+            chars.push(char::from(i));
+        }
+
+        for i in b'0'..=b'9' {
+            chars.push(char::from(i));
+        }
+
+        Alphabet { chars }
+    }
+
+    pub fn get_char(&self, client: &Client) -> Result<char> {
+        // faster to get these in bulk
+        let rand = client.get_pseudo_random(1)?;
+        let rand = rand[0] as usize % self.chars.len();
```

Comment:
get_psuedo_random(1) returns a single byte of random bits with a range of 0..=255.  a-zA-Z0-9 is 62 characters.  Since 256 % 62 != 0, need to check if the random number happens to fall into 248..=255 and throw it out if so.  Otherwise, the first 8 characters are unfairly weighted.

Reply:
that's a great catch. very much appreciate the review

Reply:
PR has been updated to account for the case where range of possible values from the PRNG is not divisible by the size of the alphabet. In this case we now drop the value from the PRNG and just get a new one.